### PR TITLE
Add TypeScript support

### PR DESCRIPTION
**Pull request by nghitran**
Add basic support for TypeScript:
[http://www.typescriptlang.org/](http://www.typescriptlang.org/)

### CODE REVIEW

File: lib/tilt/typescript.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+module Tilt
+  class TypeScriptTemplate < Template
+    self.default_mime_type = 'application/javascript'
+
+    def prepare
+    end
+
+    def evaluate(scope, locals, &block)
+      @output ||= TypeScript::Node.compile(data, '--target', 'ES5')
```
Comment (judofyr) [inline lib/tilt/typescript.rb]: Maybe these options should be exposed using `options`?

**Comment by judofyr** [comment]
The test file has a typo in it: **tile**_typescript_test.rb

**Comment by judofyr** [comment]
Something this like:

```diff
diff --git a/lib/tilt/typescript.rb b/lib/tilt/typescript.rb
index dc36a14..9d0fc1f 100644
--- a/lib/tilt/typescript.rb
+++ b/lib/tilt/typescript.rb
@@ -6,10 +6,21 @@ module Tilt
     self.default_mime_type = 'application/javascript'
 
     def prepare
+      @option_args = []
+
+      options.each do |key, value|
+        next unless value
+
+        @option_args << "--#{key}"
+
+        if value != true
+          @option_args << value.to_s
+        end
+      end
     end
 
     def evaluate(scope, locals, &block)
-      @output ||= TypeScript::Node.compile(data, '--target', 'ES5')
+      @output ||= TypeScript::Node.compile(data, *@option_args)
     end
   end
 end
diff --git a/test/tile_typescript_test.rb b/test/tile_typescript_test.rb
deleted file mode 100644
index 51fd550..0000000
--- a/test/tile_typescript_test.rb
+++ /dev/null
@@ -1,29 +0,0 @@
-require 'test_helper'
-require 'tilt'
-
-begin
-  require 'tilt/typescript'
-
-  class TypeScriptTemplateTest < Minitest::Test
-    def setup
-      @ts = "var x:number = 5"
-      @js = /var x = 5;\s*/
-    end
-
-    test "is registered for '.ts' files" do
-      assert_equal Tilt::TypeScriptTemplate, Tilt['test.ts']
-    end
-
-    test "compiles and evaluates the template on #render" do
-      template = Tilt::TypeScriptTemplate.new { @ts }
-      assert_match @js, template.render
-    end
-
-    test "can be rendered more than once" do
-      template = Tilt::TypeScriptTemplate.new { @ts }
-      3.times { assert_match @js, template.render }
-    end
-  end
-rescue LoadError => boom
-  warn "Tilt::TypeScriptTemplate (disabled)"
-end
diff --git a/test/tilt_typescript_test.rb b/test/tilt_typescript_test.rb
new file mode 100644
index 0000000..f296a0e
--- /dev/null
+++ b/test/tilt_typescript_test.rb
@@ -0,0 +1,34 @@
+require 'test_helper'
+require 'tilt'
+
+begin
+  require 'tilt/typescript'
+
+  class TypeScriptTemplateTest < Minitest::Test
+    def setup
+      @ts = "var x:number = 5"
+      @js = /var x = 5;\s*/
+    end
+
+    test "is registered for '.ts' files" do
+      assert_equal Tilt::TypeScriptTemplate, Tilt['test.ts']
+    end
+
+    test "compiles and evaluates the template on #render" do
+      template = Tilt::TypeScriptTemplate.new { @ts }
+      assert_match @js, template.render
+    end
+
+    test "supports source map" do
+      template = Tilt::TypeScriptTemplate.new(inlineSourceMap: true)  { @ts }
+      assert_match /sourceMappingURL/, template.render
+    end
+
+    test "can be rendered more than once" do
+      template = Tilt::TypeScriptTemplate.new { @ts }
+      3.times { assert_match @js, template.render }
+    end
+  end
+rescue LoadError => boom
+  warn "Tilt::TypeScriptTemplate (disabled)"
+end
```

**Comment by nghitran** [comment]
Cool! Thanks @judofyr, I updated the PR as your suggestion

**Comment by judofyr** [comment]
Sweet! Thanks for the PR.

**Pull request merged**